src/main/java/uk/ac/cam/cl/dtg/segue/database/PostgresSqlDb.java

```diff
@@ -86,4 +86,20 @@ public void close() {
         }
 
     }
+
+    /**
+     * Check whether the database is a read only replica or not.
+     * @return whether the database is read only.
+     */
+    public boolean isReadOnlyReplica() {
+        try (Connection conn = getDatabaseConnection()) {
+            PreparedStatement pst;
+            pst = conn.prepareStatement("SELECT pg_is_in_recovery()");
+            ResultSet results = pst.executeQuery();
+            results.next();
+            return results.getBoolean(1);
+        } catch (SQLException e) {
```

Comment:
I'd be tempted to say that a 500 might be more informative in the case of an exception?

Reply:
Ignore that, it is an internal method, it does not return a HTTP response. I'd still be tempted to pass on the exception and letting the calling method deciding how to handle it.

Reply:
Well, this isn't called by and HTTP code so Internal Server Error doesn't help here. But also you can get an SQL exception if the database is readonly or the query interferes with recovery, so in those cases it is true.
I'd make it throw if you felt it was important that it did?

Reply:
I guess the more correct version would then catch the specific read-only exceptions (using the union types notation stuff!) to return false and the others are left uncaught. But, you're right, probably not important!

Reply:
That would work if those exceptions corresponded to Java exceptions, but the JDBC driver only throws them all as `SQLException` and I'd have to process the contained Postgres error to work out what it was. Just throwing the exception will do for now.